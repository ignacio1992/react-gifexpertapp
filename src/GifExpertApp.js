import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGird } from './components/GifGird';

//forma basica de creacion de componentes
export const GifExpertApp = () => {

    //const categories =['One Punch',' Samuria X','Dragon Ball'];
    const [categories, setCategories] = useState(['One Punch','pokemon']);

    //const handleAdd = ()=>{
        
        //setCategories ( [...categories, 'HunterXHunter'] ); //forma de agregar contenido a un arreglo
        // al poner el arreglo de esta manera: ( ['HunterXHunter'], ...categories ); HunterXHunter quedaria primo
        //y el contenido original del arreglo le seguiria
        
        //otra forma
        // setCategories(anime =>[...anime,'HunterXhunter']);// recuerda bien los 3 puntos
        //estos agregan contenido a un arreglo. pinche pelotudo.
   // }

    return (
        <>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={ setCategories }/>
        <hr/>
        <ol>
             {
                 categories.map( category => 
                   // <li key= {category} > {category} </li>
                   <GifGird  key = {category}
                   category = { category }/> 
                )
             }
        </ol>
        </>
    )
}



