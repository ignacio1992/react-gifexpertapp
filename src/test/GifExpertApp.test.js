
import React from "react"
import { GifExpertApp } from "../GifExpertApp"
import '@testing-library/jest-dom';
import { shallow } from "enzyme";



describe('pruebas sobre el <GifExpertApp/>', () => {

  
    
    test('debe mostrar el contenido de <GifExpertApp/> ', () => {

    const wrapper = shallow (< GifExpertApp />);

    expect(wrapper).toMatchSnapshot();
        
    })
    
    test('debe de motrar una lista de categorias ', () => {
        
        const categories = ['One Punch', 'HunterXHunter']
        const wrapper = shallow (< GifExpertApp defaultCategories ={ categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGird').length).toBe( categories.length );



    })
})


