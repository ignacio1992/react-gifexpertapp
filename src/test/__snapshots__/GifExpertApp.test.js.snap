// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas sobre el <GifExpertApp/> debe de motrar una lista de categorias  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGird
      category="One Punch"
      key="One Punch"
    />
    <GifGird
      category="HunterXHunter"
      key="HunterXHunter"
    />
  </ol>
</Fragment>
`;

exports[`pruebas sobre el <GifExpertApp/> debe mostrar el contenido de <GifExpertApp/>  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
